### ISSUE TITLE
Pictus thumbnail DLL doesn't close PSD handles

### ISSUE THREAD

**Issue by mighty-cthulhu** [poppeman/Pictus#75]
OS: Windows 10

To reproduce:

- open a folder with some PSD files
- wait until thumbnails are shown
- attempt to edit or delete one of the files

Actual result: the file can't be deleted or modified until dllhost.exe dies

Expected result: the file can be deleted or modified immediately after Pictus is done generating its thumbnail

**Comment by poppeman** [comment]
The thumbnail handler doesn't open the files manually, but instead gets pre-opened streams from Windows. Does it only happen with PSD files for you?

**Comment by mighty-cthulhu** [comment]
Yes, this seems to only affect PSD files. I copied a bunch of images in different formats to a directory, opened it with Windows Explorer and then looked at the list of dllhost's file handles. The only handle it had in this directory was to the PSD file.

**Comment by poppeman** [comment]
I don't manage to reproduce that with my test files, so it might be something in those files that causes it. Are you able to send me a few of those files?

**Comment by mighty-cthulhu** [comment]
Could have something to do with the number of layers. I will send you an example file when I get home.

**Comment by mighty-cthulhu** [comment]
The issue appears to be trickier than I thought. While Pictus thumbnail DLL always keeps open handles to all PSD files, it doesn't always affect user's ability to delete and modify those files.

I tried to reproduce this issue today with a bunch of PSDs, some of which were real photos, and some test images. The thumbnail process kept handles to all PSDs. I could however still edit and delete the PSD files, even those with which I previously had problems.

In my test directory I also had a bunch of JPG files and I noticed that the thumbnail process opened those files as well, but then immediately closed them.

Here's one of my test files (in a zip archive because github does not allow to upload PSDs): [pictus-test-blank.psd.zip](https://github.com/poppeman/Pictus/files/835568/pictus-test-blank.psd.zip)
